app/src/main/java/com/xmartlabs/template/controller/SharedPreferenceController.java

```diff
@@ -0,0 +1,122 @@
+package com.xmartlabs.template.controller;
+
+import android.content.SharedPreferences;
+import android.support.annotation.CheckResult;
+import android.support.annotation.NonNull;
+
+import com.annimon.stream.Exceptional;
+import com.annimon.stream.Objects;
+import com.annimon.stream.Optional;
+import com.google.gson.Gson;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import timber.log.Timber;
+
+/**
+ * Manage the share preference storage of the application
+ *
+ * The entities will be stored in the {@link SharedPreferences} as a String.
+ * Thus, the entities objects are serialized using {@link Gson}.
+ * The entities are retrieved once from disk and then kept in memory for faster access.
+ */
+public class SharedPreferenceController extends Controller {
```

Comment:
Do you mean in the key? Could be interesting, but the problem here is that if you rename the class you lost the entity.